pkg/micro-ecc/psa_uecc/p192.c

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2021 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_psa_crypto pkg_micro-ecc
+ * @{
+ *
+ * @brief       Glue code translating between PSA Crypto and the Micro-ECC APIs
+ *
+ * @author      Lena Boeckmann <lena.boeckmann@haw-hamburg.de>
+ *
+ * @}
+ */
+
+#include "psa/crypto.h"
+#include "uECC.h"
+
+psa_status_t psa_generate_ecc_p192r1_key_pair(  const psa_key_attributes_t *attributes,
+                                                uint8_t *priv_key_buffer, uint8_t *pub_key_buffer,
+                                                size_t *priv_key_buffer_length,
+                                                size_t *pub_key_buffer_length)
+{
+    int ret = 0;
+
+    *priv_key_buffer_length = PSA_BITS_TO_BYTES(attributes->bits);
+    *pub_key_buffer_length = PSA_EXPORT_PUBLIC_KEY_OUTPUT_SIZE(attributes->type, attributes->bits);
+
+    struct uECC_Curve_t *curve = (struct uECC_Curve_t *)uECC_secp192r1();
```

Comment:
Why is all this casting needed? For what I see `uECC_secp192r1` returns type `uECC_Curve`, which is also the type for `uECC_make_key`, right?

https://github.com/kmackay/micro-ecc/blob/master/uECC.h

Reply:
You're right, it should work without casting